include/broker/detail/monotonic_buffer_resource.hh

```diff
@@ -11,7 +11,7 @@ namespace broker::detail {
 class monotonic_buffer_resource {
 public:
   monotonic_buffer_resource() {
-    allocate_block(nullptr);
+    allocate_block(nullptr, 0);
```

Comment:
> Also, if the first object allocated is more than `block_size`, this wastes one block of memory.

True, but for the use cases we have, Broker will always have something "small" to write first. In the `variant` branch also (were it blew up).

We can't just drop this first allocation, because `allocate` assumes that `current_` is always valid. So I'd have at least to add some `if`-guard there (and maybe in other places). Want me to change it?

Reply:
> We can't just drop this first allocation, because allocate assumes that current_ is always valid. So I'd have at least to add some if-guard there (and maybe in other places). Want me to change it?

No worries, no need to change anything. Thought if `_remaining = 0` then `std::align` would do the "right" thing, but didn't look further.